### DISCUSSION TITLE
Getting title of an audio disk

### DISCUSSION THREAD

**Discussion by SquidDev** [cc-tweaked/CC-Tweaked#1828] [Q&A]
I'm working on writing a music disc player with a monitor that can select discs, and the monitor needs to know the name of the disc.

One problem I'm having is with minecrafts vanilla music disc, "5", which is by Samuel Åberg. The `Å` is giving me a bit of trouble, but I thought I'd just look at the ID of the disc currently present in the drive and write the name out myself.

Unfortunately drives don't seem to have a `getItemDetail(slot)` like `inventory` types do. I can't get back `{["item"] = "minecraft:music_disc_5"}`, the only thing available through the disc player is `getDiskID()` which seems to be blank for music discs, or the `drive.getAudioTitle()`

And then I had an issue with a mod that adds a few music discs where the `drive.getAudioTitle()` just returns a translation key due to some missing translations. I'd also like that to read using the item id of the disc, but can't access it while within the player.

I could use an inventory to `pullItems()` the disc out of the player to read the id, but it'd stop the music disc that way

---

_Originally posted by @GStefanowich in https://github.com/cc-tweaked/CC-Tweaked/issues/1699#issuecomment-2103445484_

**Comment by SquidDev** [comment]
Ahh, yep. `getDiskID` is only defined for CC's floppy disks (which returns the integer id of the track), and not record discs.

> And then I had an issue with a mod that adds a few music discs where the `drive.getAudioTitle()` just returns a translation key due to some missing translations. 

Hrmr, this may not be helped by how CC gets the record's name. I've made some changes in c7e49d1929b02f5520fe38d0289be625d4f0dc6a which might improve that.

**Comment by GStefanowich** [comment]
Is there any chance of `drive`s get a method similar to `inventory` types `getItemDetail()`?

----

https://github.com/cc-tweaked/CC-Tweaked/blob/c7e49d1929b02f5520fe38d0289be625d4f0dc6a/projects/common/src/main/java/dan200/computercraft/shared/media/items/RecordMedia.java#L29-L32

Hopefully the change is good. I'm assuming `getDisplayName` on `RecordItem` Overrides and returns the disc name instead of just `Music Disc`

The mod specifically I was coming across is Tropicraft

I just tested in a vanilla jukebox and the playing text works normally
![image](https://github.com/cc-tweaked/CC-Tweaked/assets/7873930/5f0609fc-2037-47d2-887c-4c375f0ef850)

But with the computer it shows the translation key
![image](https://github.com/cc-tweaked/CC-Tweaked/assets/7873930/4c262d16-2c01-46cd-b6ea-af803b62bd9c)

Which I see `+ ".desc"` was taken out in that change
